src/core/unittest/FrameTest.cpp

```diff
@@ -243,6 +243,60 @@ TEST(FrameTest, ReliableResetStreamFrameEncodeDecode)
     ASSERT_EQ(Frame.ReliableSize, DecodedFrame.ReliableSize);
 }
 
+// 
+// Tests if the reordering threshold has been hit. This function initializes a
+// QUIC_ACK_TRACKER and populates it with the provided packet numbers. It then 
+// calls QuicAckTrackerDidHitReorderingThreshold to determine if the reordering
+// threshold has been hit.
+// 
+// 
+bool TestReorderingThreshold(
+    uint8_t ReorderingThreshold,
+    uint64_t LargestPacketNumberAcknowledged, 
+    const std::vector<std::vector<int>>& AckTrackerLists) 
+{
+    QUIC_ACK_TRACKER Tracker;
+    QuicAckTrackerInitialize(&Tracker);
+    for (const auto& List : AckTrackerLists) {
+        for (int i : List) {
+            QuicRangeAddValue(&Tracker.PacketNumbersToAck, i);
+        }
+    }
+    Tracker.LargestPacketNumberAcknowledged = LargestPacketNumberAcknowledged;
+    BOOLEAN Result = QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold);
+    QuicAckTrackerUninitialize(&Tracker);
+    return Result;
+}
+
+TEST(FrameTest, TestQuicAckTrackerDidHitReorderingThreshold)
+{
+    ASSERT_FALSE(TestReorderingThreshold(0, 0, {{100}}));
+
+    // Case 1
+    ASSERT_FALSE(TestReorderingThreshold(3, 0, {{0}}));
+    ASSERT_FALSE(TestReorderingThreshold(3, 0, {{0, 1}}));    
+    ASSERT_FALSE(TestReorderingThreshold(3, 0, {{0, 1}, {3}}));
+    ASSERT_FALSE(TestReorderingThreshold(3, 0, {{0, 1}, {3, 4}}));
+    ASSERT_TRUE(TestReorderingThreshold(3, 0, {{0, 1}, {3, 4, 5}}));
+    ASSERT_FALSE(TestReorderingThreshold(3, 5, {{0, 1}, {3, 4, 5}, {8}}));
+    ASSERT_TRUE(TestReorderingThreshold(3, 5, {{0, 1}, {3, 4, 5}, {8, 9}}));
+    ASSERT_TRUE(TestReorderingThreshold(3, 9, {{0, 1}, {3, 4, 5}, {8, 9, 10}}));
```

Comment:
A thought: Having separate arrays for each subrange doesn't seem super necessary, unless each sub array is actually min max. So, we should use one of the following instead:
```c
ASSERT_TRUE(TestReorderingThreshold(3, 9, {{0, 1}, {3, 5}, {8, 10}}));
```
or
```c
ASSERT_TRUE(TestReorderingThreshold(3, 9, {0, 1, 3, 4, 5, 8, 9, 10}));
```
I think the first is easier to read.

Reply:
Feel free to say 'Won't Fix' though.